webui-src/app/people/people_ownids.js

```diff
@@ -1,213 +1,399 @@
+                    pseudonimous,
+                    pgpPassword,
+                  },
+                  (data) => {
+                    const message = data.retval
+                      ? "Successfully created identity."
+                      : "An error occured while creating identity.";
+                    widget.popupMessage([
+                      m("h3", "Create new Identity"),
+                      m("hr"),
+                      message,
+                    ]);
+                  }
+                );
+          },
+        },
+        "Create"
+      ),
+    ],
+  };
+};
+
+const SignedEditIdentity = () => {
+  let passphase = "";
+  return {
+    view: (v) => [
+      m("i.fas.fa-user-edit"),
+      m("h3", "Enter your passpharse"),
+      m("hr"),
+
+      m("input[type=password][placeholder=Passpharse]", {
+        style: "margin-top:50px;width:80%",
+        oninput: (e) => {
+          passphase = e.target.value;
+        },
+      }),
+      m(
+        "button",
         {
+          style: "margin-top:160px;",
           onclick: () =>
             rs.rsJsonApiRequest(
-              '/rsIdentity/createIdentity',
+              "/rsIdentity/updateIdentity",
               {
-                name,
-                pseudonimous,
-                pgpPassword,
+                id: v.attrs.details.mId,
+                name: v.attrs.name,
+                pseudonimous: checksudo(v.attrs.details.mPgpId),
+                pgpPassword: passphase,
               },
-              data => {
+              (data) => {
                 const message = data.retval
-                  ? 'Successfully created identity.'
-                  : 'An error occured while creating identity.';
+                  ? "Successfully created identity."
+                  : "An error occured while creating identity.";
                 widget.popupMessage([
-                  m('h3', 'Create new Identity'),
-                  m('hr'),
+                  m("h3", "Create new Identity"),
+                  m("hr"),
                   message,
                 ]);
-                m.redraw();
-              },
+              }
             ),
         },
-        'Create',
+        "Enter"
       ),
     ],
   };
 };
 
 const EditIdentity = () => {
+  let name = "";
   return {
-    view: () => [
-      m('i.fas.fa-user-edit'),
-      m('h3', 'Edit Identity'),
-      m('hr'),
-      m('input[type=text][placeholder=Name]', {}),
-      m('canvas'),
-      m('button', 'Save'),
+    view: (v) => [
+      m("i.fas.fa-user-edit"),
+      m("h3", "Edit Identity"),
+      m("hr"),
+      m("input[type=text][placeholder=Name]", {
+        value: name,
+        oninput: (e) => {
+          name = e.target.value;
+        },
+      }),
+      m("canvas"),
+      m(
+        "button",
+        {
+          onclick: () => {
+            !checksudo(v.attrs.details.mPgpId)
+              ? widget.popupMessage([
+                  m(SignedEditIdentity, {
```

Comment:
We can try the same here.

Having a single component that handles both signed and unsigned depending on the `pseudonimous` value.